src/ralph/models/selector.py

```diff
@@ -106,25 +107,26 @@ def get_decision_tree(self, model_rules):
         if not rule_counter:
             return {}
 
+        # We retrieve the most occurred rule and use it to split the decision tree in two:
+        # 1 - true_subtree: a subtree of models that require the most occurred rule to be true
+        # 2 - false_subtree: a subtree of models that don't use the most occurred rule
         root_rule = rule_counter.most_common(1)[0][0]
```

Comment:
Edge case: what if two (or more) rules have the same number of occurrences?

Reply:
Good point, indeed this is not covered.
I ran into it just now trying to use sets instead of lists for the rules as there is a lot of "rule" in "rule_list" usage.
With sets it's undefined behavior (they are not ordered) - tests some times pass - sometimes not.
With lists the order is determined by the order we create the `model_rules`

Reply:
> Good point, indeed this is not covered.
> I ran into it just now trying to use sets instead of lists for the rules as there is a lot of "rule" in "rule_list" usage.
> With sets it's undefined behavior (they are not ordered) - tests some times pass - sometimes not.
> With lists the order is determined by the order we create the `model_rules` (not 100%, I will have to check this)

I was wrong :sweat_smile: the order will be determined by the hash value of the rule, [example](https://trinket.io/python/ad3599ed63) and [docs](https://docs.python.org/3/library/collections.html#collections.Counter)

I will retry using sets instead, just to see why it doesn't work.
Is there any action to be taken in this edge case ?

Reply:
:facepalm: The example was in python2, python3 does use the order of insertion, that's why using sets didn't worked for me. [Example](https://trinket.io/python3/77bfdc2e26)